v2/internal/attacktechniques/aws/lateral-movement/ssm-start-session/main.go

```diff
@@ -0,0 +1,81 @@
+package aws
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	"github.com/datadog/stratus-red-team/v2/pkg/stratus"
+	"github.com/datadog/stratus-red-team/v2/pkg/stratus/mitreattack"
+	"strings"
+)
+
+//go:embed main.tf
+var tf []byte
+
+func init() {
+	const codeBlock = "```"
+	stratus.GetRegistry().RegisterAttackTechnique(&stratus.AttackTechnique{
+		ID:           "aws.lateral-movement.ssm-start-session",
+		FriendlyName: "Usage of AWS Systems Manager console to start a session on multiple instances",
+		IsSlow:       true,
+		Description: `
+Simulates an attacker utilizing AWS Systems Manager (SSM) StartSession to gain unauthorized interactive access to multiple EC2 instances.
+
+Warm-up:
+
+- Create multiple EC2 instances and a VPC (takes a few minutes).
+
+Detonation: 
+
+- Initiates a connection to the EC2 for a Session Manager session.
+
+References:
+
+- https://hackingthe.cloud/aws/post_exploitation/run_shell_commands_on_ec2/#session-manager
+`,
+		Detection: `
+Identify, through CloudTrail's <code>StartSession</code> event, when a user is starting an interactive session to multiple EC2 instances. Sample event:
+
+` + codeBlock + `
+{
+  "eventSource": "ssm.amazonaws.com",
+  "eventName": "StartSession",
+  "requestParameters": {
+    "target": "i-123456"
+  },
+  "responseElements": {
+        "sessionId": "...",
+        "tokenValue": "Value hidden due to security reasons.",
+        "streamUrl": "wss://ssmmessages.eu-west-1.amazonaws.com/v1/data-channel/..."
+   },
+}
+` + codeBlock + `
+`,
+		Platform:                   stratus.AWS,
+		PrerequisitesTerraformCode: tf,
+		IsIdempotent:               true,
+		MitreAttackTactics:         []mitreattack.Tactic{mitreattack.LateralMovement},
+		Detonate:                   detonate,
+	})
+}
+
+func detonate(params map[string]string, providers stratus.CloudProviders) error {
+	ssmClient := ssm.NewFromConfig(providers.AWS().GetConnection())
+	instanceIDs := strings.Split(params["instance_ids"], ",")
+
+	for _, instanceID := range instanceIDs {
+		cleanInstanceID := strings.Trim(instanceID, " \"\n\r")
+		_, err := ssmClient.StartSession(context.TODO(), &ssm.StartSessionInput{
```

Comment:
Wdyt on using `SendCommand` instead? Maybe it's worth 2 techniques, maybe not.

Reply:
@Frichetten curious if you have an opinion from an attacker's standpoint here!

Reply:
I'd lean towards having two techniques, 1 for `StartSession` and 1 for `SendCommand`. Both could be used for pivoting, stealing the IAM role of the host, persistence, etc. I have some info documented [here](https://hackingthe.cloud/aws/post_exploitation/run_shell_commands_on_ec2/) if it helps.

Reply:
thanks!

Reply:
tracked in https://github.com/DataDog/stratus-red-team/issues/480